uberfire-extensions/uberfire-widgets/uberfire-widgets-core/uberfire-widgets-core-client/src/main/java/org/uberfire/ext/widgets/core/client/tree/TreeItem.java

```diff
@@ -98,10 +132,10 @@ public void onClick(ClickEvent event) {
                 initWidget(folder);
             }
         } else if (type.equals(Type.ITEM)) {
-            this.item = GWT.create(FlowPanel.class);
+            this.item = new FlowPanel();
```

Comment:
Probably you can use here the `contentProvider` in order to avoid using the `new` operator for creating the `FlowPanel`. This means that right now you're not able to test this logic inside this `if` statement, how're you mocking it? :)